Source/control.cpp

```diff
@@ -298,20 +298,21 @@ void SetSpellTrans(spell_type t)
  */
 static void DrawSpell(const CelOutputBuffer &out)
 {
-	spell_id spl = plr[myplr]._pRSpell;
-	spell_type st = plr[myplr]._pRSplType;
+	auto &myPlayer = plr[myplr];
+	spell_id spl = myPlayer._pRSpell;
+	spell_type st = myPlayer._pRSplType;
 
 	// BUGFIX: Move the next line into the if statement to avoid OOB (SPL_INVALID is -1) (fixed)
 	if (st == RSPLTYPE_SPELL && spl != SPL_INVALID) {
-		int tlvl = plr[myplr]._pISplLvlAdd + plr[myplr]._pSplLvl[spl];
+		int tlvl = myPlayer._pISplLvlAdd + myPlayer._pSplLvl[spl];
 		if (!CheckSpell(myplr, spl, st, true))
 			st = RSPLTYPE_INVALID;
 		if (tlvl <= 0)
 			st = RSPLTYPE_INVALID;
 	}
 	if (currlevel == 0 && st != RSPLTYPE_INVALID && !spelldata[spl].sTownSpell)
 		st = RSPLTYPE_INVALID;
-	if (plr[myplr]._pRSpell < 0)
+	if (myPlayer._pRSpell < 0)

```

Comment:
This could be replaced with `if (spl < 0)`.

Reply:
A bit of a useless sanity check at this point since we applied the enums and they are defined as uint8_t so can never be less then 0